### DISCUSSION TITLE
Make a bug report guideline to guide new github users

### DISCUSSION THREAD

**Discussion by Oxey405** [4ian/GDevelop#5320] [Enhancement ideas (and their technical discussions)]
## The problem
Hello !
Throughout my browsing of the issues in GDevelop, I noticed a recurrent problem in the issues about crash reports.
The problem is that, most users don't even fill out the form and just send the crash report as it's nothing and never answer back to contributor's questions.
## Suggested solution :

Make a in-editor bug report form that sends all of the crash report data to a github action which rewrite all infos to a github issue.
Also make a clear (if it does not already exists) crash report guideline for everyone translated in as many languages as possible through crowdin.

------

To be clear, I am not putting the blame on new users but more on the fact that it's not practical for inexperimented users to understand github. 

I'm willing to make a PR for a new in-IDE screen form for that but for the "post it to github" part, I'm not quite sure how to do it because I don't have direct access to GDevelop's repo.

**Comment by Oxey405** [comment]
Update : I found a way to include all of the form data in a single Github URL :

[https://github.com/4ian/GDevelop/issues/new?assignees=&labels=&template=--bug-report.md&title=Crash while using an editor&body=[form content]](https://github.com/4ian/GDevelop/issues/new?assignees=&labels=&template=--bug-report.md&title=Crash%20while%20using%20an%20editor&body=[form%20content])

This can be clicked by the user and he does not have to fill out the crash report by himself.

**Comment by AlexandreSi** [comment]
Hi @Oxey405 !
Thanks for the idea, it's indeed a bit frustrating to see open issues about crashes without answers from the OP.

A quick explanation about the bug reporting (from a technical point of view):
- a crash is caught by the component `ErrorBoundary` https://github.com/4ian/GDevelop/blob/master/newIDE/app/src/UI/ErrorBoundary.js
- you can see that we are using the GitHub endpoint that allows to open a pre-filled issue (we are not using a GitHub issue template though)
- we are writing as much info as we can in the issue body.

So far, I personally did not thought about how to increase or improve the data that we gather about the crash. So maybe there's something here to improve.

Regarding the issue that is "How to make the user send context about the crash", this is quite complexe since it's related to human behavior:
- A crash happened, it might not have caused troubles, but it could so the user might be angry/frustrated/sad.
- When in that state of mind, one usually does not care about sending a report:
  - To click on a button might not seem too much to ask but it's usually a lot.
  - Especially if they need a GitHub account for that and that they don't have one.

Considering this situation, we can consider that we end up having 1 issue opened on GitHub for something like 10 crashes. Or is it 100? 1000? It's hard to know. I don't think we send analytics about that.

So we can add a form to fill out when a crash happens but it's very likely that we would see far less crash reports.

Maybe we could create an incentive to create beautiful crash reports with detailed context but I lack ideas about how to achieve that. What do you think?

**Comment by Oxey405** [comment]
Hey @AlexandreSi I thought a bit more about it ; 
Maybe we should just build a crash report system with GDevelop accounts as identifiers... 
It may ask a bit more than just a IDE update though. Maybe I can build a "crash report server" that also copies the reports in github issues...

What do you think ?
